cmd/imagePushToRegistry.go

```diff
@@ -0,0 +1,200 @@
+
+	err := handleCredentialsForPrivateRegistry(config.DockerConfigJSON, sourceRegistry, config.SourceRegistryUser, config.SourceRegistryPassword, utils)
+	if err != nil {
+		return errors.Wrap(err, "failed to handle credentials for source registry")
+	}
+
+	err = handleCredentialsForPrivateRegistry(config.DockerConfigJSON, targetRegistry, config.TargetRegistryUser, config.TargetRegistryPassword, utils)
+	if err != nil {
+		return errors.Wrap(err, "failed to handle credentials for target registry")
+	}
+
+	if len(config.LocalDockerImagePath) > 0 {
+		if err := pushLocalImageToTargetRegistry(config.LocalDockerImagePath, dst, config.TagLatest, utils); err != nil {
+			return errors.Wrapf(err, "failed to push local image to %q", targetRegistry)
+		}
+		return nil
+	}
+
+	if err := copyImage(src, dst, config.TagLatest, utils); err != nil {
+		return errors.Wrapf(err, "failed to copy image from %q to %q", sourceRegistry, targetRegistry)
+	}
+
+	return nil
+}
+
+func handleCredentialsForPrivateRegistry(dockerConfigJsonPath, registry, username, password string, utils imagePushToRegistryUtils) error {
+	if len(dockerConfigJsonPath) == 0 && (len(registry) == 0 || len(username) == 0 || len(password) == 0) {
+		return nil
+	}
+
+	if len(dockerConfigJsonPath) == 0 {
+		_, err := docker.CreateDockerConfigJSON(registry, username, password, "", targetDockerConfigPath, utils)
+		if err != nil {
+			return errors.Wrap(err, "failed to create new docker config")
+		}
+		return nil
+	}
+
+	_, err := docker.CreateDockerConfigJSON(registry, username, password, targetDockerConfigPath, dockerConfigJsonPath, utils)
+	if err != nil {
+		return errors.Wrapf(err, "failed to update docker config %q", dockerConfigJsonPath)
+	}
+
+	err = docker.MergeDockerConfigJSON(targetDockerConfigPath, dockerConfigJsonPath, utils)
+	if err != nil {
+		return errors.Wrapf(err, "failed to merge docker config files")
+	}
+
+	return nil
+}
+
+func copyImage(src, dst string, tagLatest bool, utils imagePushToRegistryUtils) error {
+	if tagLatest {
+		// imageName is repository + image, e.g test.registry/testImage
+		imageName, _ := parseDockerImage(dst)
+		if err := utils.CopyImage(src, imageName); err != nil {
+			return err
+		}
+	}
+
+	return utils.CopyImage(src, dst)
+}
+
+func pushLocalImageToTargetRegistry(localDockerImagePath, dst string, tagLatest bool, utils imagePushToRegistryUtils) error {
+	img, err := utils.LoadImage(localDockerImagePath)
+	if err != nil {
+		return err
+	}
+
+	if tagLatest {
+		// imageName is repository + image, e.g test.registry/testImage
+		imageName, _ := parseDockerImage(dst)
+		if err := utils.PushImage(img, imageName); err != nil {
+			return err
+		}
+	}
+
+	return utils.PushImage(img, dst)
+}
+
+func parseDockerImage(image string) (string, string) {
+	re := regexp.MustCompile(`^(.*?)(?::([^:/]+))?$`)
+
+	matches := re.FindStringSubmatch(image)
+	if len(matches) > 1 {
+		return matches[1], matches[2]
+	}
+
+	return image, ""
+}
```

Comment:
Both calls of `parseDockerImage` function explicitly omit the second return parameter. It would be better to remove it to keep things clean.